test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -728,6 +785,96 @@ public void CreateDisplayMetadata_IsFlagsEnum_ReflectsModelType(Type type, bool
         }
 
         [Fact]
+        public void CreateDisplayMetadata_EnumGroupedDisplayNamesAndValues_IStringLocalizer()
+        {
+            // Arrange
+            var provider = CreateIStringLocalizerProvider();
+
+            var key = ModelMetadataIdentity.ForType(typeof(EnumWithLocalizedDisplayNames));
+            var attributes = new object[0];
+
+            var context = new DisplayMetadataProviderContext(key, new ModelAttributes(attributes));
+            provider.CreateDisplayMetadata(context);
+
+            // Act & Assert
+            using (new CultureReplacer("fr-FR", "fr-FR"))
+            {
+                Assert.All(
+                    context.DisplayMetadata.EnumGroupedDisplayNamesAndValues
+                        .Where(kvp => kvp.Value == "2"),
+                    kvp => Assert.Equal("Loc_Two_Name fr-FR", kvp.Key.Name));
+            }
+
+            using (new CultureReplacer("en-US", "en-US"))
+            {
+                Assert.All(
+                    context.DisplayMetadata.EnumGroupedDisplayNamesAndValues
+                        .Where(kvp => kvp.Value == "2"),
+                    kvp => Assert.Equal("Loc_Two_Name en-US", kvp.Key.Name));
+            }
+        }
+
+        [Fact]
+        public void CreateDisplayMetadata_EnumGroupedDisplayNamesAndValues_NameWithOnlyDisplayAttribute()
```

Comment:
This is a unit test of `EnumGroupAndName`.

This test also covers an uninteresting scenario. Since only `ModelMetadata` stores instances of this class, `ModelMetadata.EnumGroupedDisplayNamesAndValues` could return a new collection every time it's read and MVC's behaviour would be the same.